system/Database/Postgre/Builder.php

```diff
@@ -312,6 +312,131 @@ public function join(string $table, $cond, string $type = '', ?bool $escape = nu
         return parent::join($table, $cond, $type, $escape);
     }
 
+    /**
+     * Generates a platform-specific batch update string from the supplied data
+     *
+     * @used-by batchExecute
+     *
+     * @param string                 $table  Protected table name
+     * @param list<string>           $keys   QBKeys
+     * @param list<list<int|string>> $values QBSet
+     */
+    protected function _updateBatch(string $table, array $keys, array $values): string
+    {
+        $sql = $this->QBOptions['sql'] ?? '';
+
+        // if this is the first iteration of batch then we need to build skeleton sql
+        if ($sql === '') {
+            $constraints = $this->QBOptions['constraints'] ?? [];
+
+            if ($constraints === []) {
+                if ($this->db->DBDebug) {
+                    throw new DatabaseException('You must specify a constraint to match on for batch updates.'); // @codeCoverageIgnore
+                }
+
+                return ''; // @codeCoverageIgnore
+            }
+
+            $updateFields = $this->QBOptions['updateFields'] ??
+                $this->updateFields($keys, false, $constraints)->QBOptions['updateFields'] ??
+                [];
+
+            $alias = $this->QBOptions['alias'] ?? '_u';
+
+            $sql = 'UPDATE ' . $this->compileIgnore('update') . $table . "\n";
+
+            $sql .= "SET\n";
+
+            $that = $this;
+            $sql .= implode(
+                ",\n",
+                array_map(
+                    static fn ($key, $value) => $key . ($value instanceof RawSql ?
+                            ' = ' . $value :
+                            ' = ' . $alias . '.' . $that->cast($value, $that->getFieldType($table, $key))),
```

Comment:
```suggestion
                            ' = ' . $that->cast($alias . '.' . $value, $that->getFieldType($table, $key))),
```

$alias . '.' . $value is the full part of the expression. I would like to see us use the more universal format of CAST(expression AS type). If we use this then you definitely have to include the alias as part of expression. The expression here is  "table"."column"

Reply:
Done.